src/sample_utils.py

```diff
@@ -185,10 +187,70 @@ def wait_for_anf_resource(client, resource_id, interval_in_sec=10, retries=60, r
                 )
                 
             break
-        except CloudError as ex:
+        except ResourceNotFoundError as ex:
             pass
 
 
+def wait_for_mirrored_status(client, resource_group, account_name, pool_name, volume_name, interval_in_sec=10, retries=60):
```

Comment:
>wait_for_mirrored_status [](http://example.com/codeflow?start=4&length=24)

Let's keep it as a single function, e.g. wait_for_mirror_status and include an argument for the mirror state we want to check, possible ones would be "Mirrored" or "Broken", it would be nice to have this as an enum and accept the enum like mirror_state.mirrored and mirror_state.broken so the user will have a clear idea of possible values.